include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_utils.h

```diff
@@ -688,12 +688,20 @@ class __future : private std::tuple<_Args...>
     {
         return __my_event;
     }
+
     operator _Event() const { return event(); }
+
     void
     wait()
     {
-#if !ONEDPL_ALLOW_DEFERRED_WAITING
         __my_event.wait_and_throw();
+    }
+
+    void
+    deferrable_wait()
```

Comment:
Do we want users to be encouraged to call a `deferrable_wait()` function?  I would think that is somewhat implementation details or perhaps to convoluted for public consumption.
  
Obviously we need this to be `public`, but should we uglify it maybe to discourage usage outside the library (or would we want some internal version of `__future` even which we don't return to the users?

Reply:
Our `__future` implemented in the namespace 
```C++
namespace oneapi
{
namespace dpl
{
namespace __internal
{
```
This mean (for me) that user will never works with our `__future` implementation directly.

Reply:
Actually they might, if an object of this type is returned by our public API functions. It might be documented as an unspecified type, yet it should be considered as a part of public API. Is that the case with `__future`?

Reply:
Yes, its an unspecified future-like type in our documentation, which lists `wait()` and `get()`.  It of course is not in the spec because it is part of the experimental async API.